packages/custom-functions-metadata/src/generate.ts

```diff
@@ -196,157 +206,208 @@ export async function generate(inputFile: string, outputFileName: string, wantCo
-
-                    const functionMetadata: IFunction = {
-                        description,
-                        helpUrl,
-                        id,
-                        name,
-                        options,
-                        parameters,
-                        result,
-                    };
-
-                    if (!options.cancelable && !options.requiresAddress && !options.stream && !options.volatile) {
-                        delete functionMetadata.options;
-                    } else {
-                        if (!options.cancelable) {
-                            delete options.cancelable;
-                        }
-
-                        if (!options.requiresAddress) {
-                            delete options.requiresAddress;
-                        }
-
-                        if (!options.stream) {
-                            delete options.stream;
-                        }
-
-                        if (!options.volatile) {
-                            delete options.volatile;
-                        }
-                    }
-
-                    if (!functionMetadata.helpUrl) {
-                        delete functionMetadata.helpUrl;
-                    }
-
-                    if (!functionMetadata.description) {
-                        delete functionMetadata.description;
-                    }
-
-                    if (!functionMetadata.result) {
-                        delete functionMetadata.result;
-                    }
-
-                    extras.push(extra);
-                    functions.push(functionMetadata);
-                }
+        functionNames.push(functionName);
+
+        if (isCustomFunction(functionDeclaration)) {
+          const extra: IFunctionExtras = {
+            errors: functionErrors,
+            javascriptFunctionName: functionName
+          };
+          const idName = getTagComment(functionDeclaration, CUSTOM_FUNCTION);
+          const idNameArray = idName.split(" ");
+          const jsDocParamInfo = getJSDocParams(functionDeclaration);
+          const jsDocParamTypeInfo = getJSDocParamsType(functionDeclaration);
+          const jsDocParamOptionalInfo = getJSDocParamsOptionalType(
+            functionDeclaration
+          );
+
+          const [lastParameter] = functionDeclaration.parameters.slice(-1);
+          const isStreamingFunction = hasStreamingInvocationParameter(
+            lastParameter,
+            jsDocParamTypeInfo
+          );
+          const isCancelableFunction = hasCancelableInvocationParameter(
+            lastParameter,
+            jsDocParamTypeInfo
+          );
+          const isInvocationFunction = hasInvocationParameter(
+            lastParameter,
+            jsDocParamTypeInfo
+          );
+
+          const parametersToParse =
+            isStreamingFunction || isCancelableFunction || isInvocationFunction
+              ? functionDeclaration.parameters.slice(
+                  0,
```

Comment:
yuck, if the line length is longer, would it prevent this? I wish there was a way to wrap across multiple lines but put as many parameters per line as fits.